src/garage/envs/bullet/bullet_env.py

```diff
@@ -174,10 +269,10 @@ def __setstate__(self, state):
 
         """
         env_id = state['id']
-        env_name = state['env_name']
+        max_episode_length = state['max_episode_length']
         # Create a environment via constructor arguments
         del state['id']
-        del state['env_name']
+        del state['max_episode_length']
         env = gym.make(env_id, **state)
 
-        self.__init__(env, env_name)
+        self.__init__(env=env, max_episode_length=max_episode_length)
```

Comment:
does this work if someone passed me an `env` by object instead of by string?

Reply:
yes this is tested at https://github.com/rlworkgroup/garage/blob/master/tests/garage/envs/test_garage_env.py#L38